src/roles/actors.rs

```diff
@@ -212,19 +212,29 @@ impl<T: Trait> Module<T> {
         role: Role,
         member_id: MemberId<T>,
         unbonding_period: T::BlockNumber,
+        stake: BalanceOf<T>,
     ) {
         // simple unstaking ...only applying unbonding period
-        let value = T::Currency::free_balance(&actor_account);
-        T::Currency::set_lock(
-            STAKING_ID,
+        Self::update_lock(
             &actor_account,
-            value,
+            stake,
```

Comment:
This is IMHO a little tricky.

Previously, actors staked their entire balance, so updating the lock for their balance makes some sense. Now you've updated this to apply not to the entire balance, but a chosen amount. So far, so good.

However, when you look further below where this function is called, it always passes the minimum staking balance for the role here. So consider this scenario:

- Account stakes when minimum staking balance is A.
- Role parameters change minimum staking balance to B.
- Account unstakes when minimum staking balance is B.

If A > B, the actor wouldn't get their entire amount unstaked again, which is definitely an issue. If B > A, perhaps unstaking fails completely?

I think the approach is correct, but it seems you then also have to record the exact amount staked when it is staked, and then unstake exactly that amount, no more and no less.

Reply:
As we discussed, unstaking will not because of how locking mechanism in the balances module is implemented. The lock is a limit/restriction set on how funds can be withdrawn or transferred (it doesn't actually involve moving any funds).

Reply:
Yes, that makes sense.